File: apps/web/src/test/fixtures/mqtt.ts
import type { MQTTStatus } from "@/entities/mqtt/model/types";

export const mqttConnectedFixture: MQTTStatus = {
  enabled: true,
  connected: true,
  status: "connected",
  brokerUrl: "tcp://mqtt:1883",
  clientId: "smr-simulation-publisher",
  topicPrefix: "smr/site-001/unit-001",
  qos: 0,
  retain: false,
  publishIntervalMs: 1000,
  lastConnectedAt: "2026-05-21T06:00:00Z",
  lastSuccessfulPublishAt: "2026-05-21T06:00:10Z",
  messagesPublished: 42,
  messagesFailed: 0,
  simulationOnly: true,
  safetyDisclaimer: "MQTT topics contain synthetic simulation payloads only.",
};

export const mqttDegradedFixture: MQTTStatus = {
  ...mqttConnectedFixture,
  connected: false,
  status: "degraded",
  lastErrorAt: "2026-05-21T06:01:00Z",
  lastErrorMessage: "broker unavailable",
  messagesFailed: 2,
};

File: apps/simulation/internal/model/mqtt.go
package model

import "time"

type MQTTStatus struct {
	Enabled                 bool       `json:"enabled"`
	Connected               bool       `json:"connected"`
	Status                  string     `json:"status"`
	BrokerURL               string     `json:"brokerUrl"`
	ClientID                string     `json:"clientId"`
	TopicPrefix             string     `json:"topicPrefix"`
	QoS                     int        `json:"qos"`
	Retain                  bool       `json:"retain"`
	PublishIntervalMS       int        `json:"publishIntervalMs"`
	LastConnectedAt         *time.Time `json:"lastConnectedAt,omitempty"`
	LastDisconnectedAt      *time.Time `json:"lastDisconnectedAt,omitempty"`
	LastSuccessfulPublishAt *time.Time `json:"lastSuccessfulPublishAt,omitempty"`
	LastErrorAt             *time.Time `json:"lastErrorAt,omitempty"`
	LastErrorMessage        string     `json:"lastErrorMessage,omitempty"`
	MessagesPublished       uint64     `json:"messagesPublished"`
	MessagesFailed          uint64     `json:"messagesFailed"`
	SimulationOnly          bool       `json:"simulationOnly"`
	SafetyDisclaimer        string     `json:"safetyDisclaimer"`
}
